src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -280,6 +281,30 @@ public Response getDataset(@PathParam("datasetId") Integer datasetId){
         }
     }
 
+    @GET
+    @Path("/batch")
+    @Produces("application/json")
+    @PermitAll
+    public Response getDatasets(@QueryParam("ids") List<Integer> datasetIds){
+        try {
+            List<Dataset> datasets = datasetService.getDatasets(datasetIds);
+
+            List<Integer> foundIds = datasets.stream().map(Dataset::getDataSetId).toList();
+            if (!foundIds.containsAll(datasetIds)) {
+                // find the differences
+                List<Integer> differences = new ArrayList<>(datasetIds);
+                differences.removeAll(foundIds);
+                throw new NotFoundException(
+                        "Could not find datasets with ids: "
+                                + String.join(",", differences.stream().map((i) -> i.toString()).collect(Collectors.toSet())));
+
+            }
+            return Response.ok(datasets, MediaType.APPLICATION_JSON).build();
```

Comment:
One more thing, this can be simpler, `MediaType.APPLICATION_JSON` is already specified on line 286.

```
return Response.ok(datasets).build();
```
or 
```
return Response.ok().entity(datasets).build();
```
In the above cases, Jackson will coerce the objects to JSON format based on `@Produces`